### PR TITLE
bullet-featherstone: support off-diagonal inertia

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Allows loading models with off-diagonal inertia terms in bullet-featherstone

## Summary

Currently gz-physics prints error messages and fails to load portions of models that contain links with off-diagonal inertia terms with magnitude larger than `0.001`.

The bullet APIs expect the moment of inertia matrix to already be diagonalized. This changes the bullet-featherstone plugin to compute the principal moments of inertia in the same manner as the bullet plugin (see [bullet's SDFFeatures.cc](https://github.com/gazebosim/gz-physics/blob/gz-physics6/bullet/src/SDFFeatures.cc#L148-L152)). It will now load models with off-diagonal inertia values without giving errors.

I'd really like to add a test for this, but I'm having trouble figuring out a simple test.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by scpeters** [comment]
I've been working on a test for this, and while I'm not ready to add one to this branch, I've done enough testing that I'm confident enough about this change to merge once this has an approval. I will post the branches I have been using for testing once I have a chance to clean them up.

cc @iche033

**Review by iche033 — APPROVED** [review]

**Pull request merged**
